dapr-spring/dapr-spring-boot-autoconfigure/src/main/java/io/dapr/spring/boot/autoconfigure/statestore/DaprStateStoreAutoConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.spring.boot.autoconfigure.statestore;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.dapr.client.DaprClient;
+import io.dapr.client.DaprClientBuilder;
+import io.dapr.spring.boot.autoconfigure.client.DaprClientAutoConfiguration;
+import io.dapr.spring.data.DaprKeyValueAdapterResolver;
+import io.dapr.spring.data.DaprKeyValueTemplate;
+import io.dapr.spring.data.KeyValueAdapterResolver;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.data.keyvalue.core.KeyValueAdapter;
+
+@AutoConfiguration(after = DaprClientAutoConfiguration.class)
+@ConditionalOnClass({DaprClient.class, KeyValueAdapter.class})
+@EnableConfigurationProperties(DaprStateStoreProperties.class)
+public class DaprStateStoreAutoConfiguration {
```

Comment:
@ThomasVitale @artur-ciocanu I am starting to think that we don't need this autoconfiguration. Mostly because if you look at Spring Data, you will see that users need to basically provide a @configuration class that creates the instances of the KeyValueAdapterResolver and the KeyValueTempplate -> https://docs.spring.io/spring-data/keyvalue/reference/keyvalue/template.html . So maybe in our attempt to make things easier, we made it more complicated for ourselves. As soon as we have in our Docs that whenever you want a DaprKeyValue template, you just need to provide a @Configuration class to get the instances based on the DaprBuilder and the DaprStateProperties. What do you think folks?

Reply:
As I mentioned in a previous comment, I have moved the `org.springframework.boot.autoconfigure.AutoConfiguration.imports` to starter.

Reply:
Hmm.. I am not sure what this means.. because I think we don't even need the AutoConfigurations.. Can you share the link of where you moved things? I need more context to understand what you meant by:
```
 have moved the org.springframework.boot.autoconfigure.AutoConfiguration.imports to starter.
```